### ISSUE TITLE
Registration | Username validation issue 

### ISSUE THREAD

**Issue by hernandez-jordan** [strapi/strapi#8847]
## Bug report

extensions/users-permissions/models/User.settings.json
```json
"attributes": {
    "username": {
      "type": "string",
      "minLength": 3,
      "unique": true,
      "configurable": true,
      "required": true
    },
    "email": {
      "type": "email",
      "minLength": 6,
      "configurable": true,
      "required": true
    },
    "password": {
      "type": "password",
      "minLength": 6,
      "configurable": true,
      "private": true
    },
```
### Describe the bug

Registration of user with an empty username field gets passed the validation.

### Steps to reproduce the behavior

1. Go to to your graphql playground
2. Create a register mutation for user with the attributes: username, password and email
3. Register user with an empty username field
4. See error: fails to validate the username with empty field. The empty **password** and **email** fields **does** throw a **validation error** though, if recreated in the same way. So the issue here is just the username empty field validation.

### Expected behavior

An error for an empty username field or minLength of 3

### Screenshots
**graphql playground**
![Screenshot 2020-12-09 at 14 27 56](https://user-images.githubusercontent.com/54027433/101635892-dca88080-3a2a-11eb-88fb-db3f2b1e03f8.png)

**Registered users**
![Screenshot 2020-12-09 at 14 47 30](https://user-images.githubusercontent.com/54027433/101637837-90127480-3a2d-11eb-9aed-ba05e2fc50b5.png)

### System

- Node.js version: v14.13.0
- Yarn version: 1.22.5
- Strapi version: 3.3.3
- Database: mongoDB
- Operating system: macOS bigSur 11.0.1

**Comment by clevercodenl** [comment]
I've the same issue

**Comment by wevisualizeit** [comment]
@soupette @petersg83 Could you help me with this issue ?

**Comment by clevercodenl** [comment]
@wevisualizeit maybe it could help if you add a proper label and actually asign the issue to one of the official Strapi developers

**Comment by wevisualizeit** [comment]
Username, email and password are required fields. How is not filling one in an "expected behaviour"? Can you elaborate on that please?

**Comment by brettus78** [comment]
> Username, email and password are required fields. How is not filling one in an "expected behaviour"? Can you elaborate on that, please?

Apologies to confuse, I was asking the question as I just noted previously that 'identifier' is either username OR email address and wondered if this was possibly expected behaviour. My mistake.
Rolled an instance to check and I agree, this is a bug.
Tried register without a username and returned:
``` "message": "Field \"UsersPermissionsRegisterInput.username\" of required type \"String!\" was not provided.",```
Tried register with username as "" (same as you) and get the same result as you (user created in spite of empty string provided).
```
mutation {
  register(input: {email:"xxxxx@xxx.com", username:"", password:"testing123"})
  {
    user{
      id
      username
      email
    }
  }
}
```

```
{
  "data": {
    "register": {
      "user": {
        "id": "35",
        "username": "",
        "email": "xxxxx@xxx.com"
      }
    }
  }
}
```

**Comment by derrickmehaffy** [comment]
> @wevisualizeit maybe it could help if you add a proper label and actually asign the issue to one of the official Strapi developers

@clevercodenl Never set the label or any assigners, we the Strapi team will do that.

**Comment by clevercodenl** [comment]
@derrickmehaffy yes understood

**Comment by vdilly** [comment]
This is still an issue with Strapi V4.0.6

**Comment by alexandrebodin** [comment]
This was fixed in v4.1.10. Going to close this
